src/main/scala/viper/server/core/ViperCoreServer.scala

```diff
@@ -0,0 +1,79 @@
+package viper.server.core
+
+import akka.actor.ActorRef
+import viper.server.ViperConfig
+import viper.server.core.ViperBackendConfigs._
+import viper.server.vsi.{Letter, ProcessManagement, VerificationJobHandler}
+import viper.silver.ast
+import viper.silver.logger.ViperLogger
+
+import scala.language.postfixOps
+
+class SilverLetter() extends Letter
+
+class ViperCoreServer(private var _config: ViperConfig) extends ProcessManagement {
+
+  // --- VCS : Configuration ---
+
+  final def config: ViperConfig = _config
+
+  protected var _logger: ViperLogger = _
+  final def logger: ViperLogger = _logger
+
+
+  /** Configures an instance of ViperCoreServer.
+    *
+    * This function should be called before any other.
+    * */
+  def start(): Unit = {
+    println("Hi from VCS start")
+    config.verify()
+
+    _logger = ViperLogger("ViperServerLogger", config.getLogFileWithGuarantee, config.logLevel())
+    println(s"Writing [level:${config.logLevel()}] logs into ${if (!config.logFile.isSupplied) "(default) " else ""}journal: ${logger.file.get}")
+
+    ViperCache.initialize(logger.get, config.backendSpecificCache())
+
+    println(s"ViperServer online in CoreServer mode")
+    super.start(config.maximumActiveJobs())
+  }
+
+  /** Verifies a Viper AST using the specified backend.
+    * */
+  def verify(programID: String, backend_config: ViperBackendConfig, program: ast.Program): VerificationJobHandler = {
+    val args: List[String] = backend_config match {
+      case _ : SiliconConfig => "silicon" :: backend_config.partialCommandLine
+      case _ : CarbonConfig => "carbon" :: backend_config.partialCommandLine
+      case _ : CustomConfig => "DummyFrontend" :: backend_config.partialCommandLine
+    }
+    val task_backend = new VerificationWorker(_config, logger.get, args :+ programID, program)
+    initializeVerificationProcess(task_backend)
+  }
+
+  /** Stops an instance of ViperCoreServer from running.
+    *
+    * As such it should be the ultimate method called. Calling any other function after 'stop()' will result in an
+    * IllegalStateException.
+    * */
+  override def stop(): Unit = {
+    super.stop()
+    println(s"Stopping ViperCoreServer")
+  }
+
+  def flushCache(): Unit = {
+    if(!isRunning) {
+      throw new IllegalStateException("Instance of ViperCoreServer already stopped")
+    }
+    ViperCache.resetCache()
+    println(s"The cache has been flushed successfully.")
```

Comment:
Please take care of the ```println```s. You may leave logging in the code, but not ```println```, as the client may want to keep their terminal clean.